lib/ansible/plugins/strategy/free.py

```diff
@@ -120,14 +120,17 @@ def run(self, iterator, play_context):
                 (state, task) = iterator.get_next_task_for_host(host, peek=True)
                 display.debug("free host state: %s" % state, host=host_name)
                 display.debug("free host task: %s" % task, host=host_name)
-                if host_name not in self._tqm._unreachable_hosts and task:
 
+                # check if there is work to do, either there is a task or the host is still blocked which could
+                # mean that it is processing an include task and after its result is processed there might be
+                # more tasks to run
+                if task or self._blocked_hosts.get(host_name, False):
```

Comment:
Looks good to me. Could an unreachable host end up in blocked hosts? If so, this would get the debug message with more work to do, but there's be no conclusory "skipping" debug output.
If that could happen, this could be:
```python
if (task or self._blocked_hosts.get(host_name, False)) and host_name not in self._tqm._unreachable_hosts:
```

Reply:
Good catch. 👍  Fixed.